cmd/apprepository-controller/controller.go

```diff
@@ -582,10 +583,14 @@ func apprepoSyncJobEnvVars(apprepo *apprepov1alpha1.AppRepository) []corev1.EnvV
 		},
 	})
 	if apprepo.Spec.Auth.Header != nil {
+		secretKeyRef := apprepo.Spec.Auth.Header.SecretKeyRef
+		if apprepo.ObjectMeta.Namespace != kubeappsNamespace {
+			secretKeyRef.LocalObjectReference.Name = kube.KubeappsSecretNameForRepo(apprepo.ObjectMeta.Name, apprepo.ObjectMeta.Namespace)
+		}
```

Comment:
I wonder if it would be simpler to just create a copy of the apprepo in the kubeapps namespace. That would help with the garbage collection and wouldn't require to change these secret references on the fly

Reply:
It would, if we weren't transitioning towards an end goal of syncing across all namespaces and running the jobs in those namespaces. This way we can begin syncing AppRepos across namespaces, we are just still running the sync jobs in the kubeapps namespace until we can update the assetsvc to receive data from the syncer.

Thanks!